src/mlpack/core/math/random.hpp

```diff
@@ -162,11 +145,26 @@ inline double RandNormal()
  * @param mean Mean of distribution.
  * @param stddev Standard deviation of distribution.
  */
-inline double RandNormal(const double mean, const double stddev)
-{
+inline double RandNormal(const double mean, const double stddev) {
   return stddev * RandNormalDist()(RandGen()) + mean;
 }
 
+/**
+ * Generates random samples from a categorical distribution based
+ *
+ * @param weights arma::Col<double> of weights summing to 1.
+ */
+inline size_t RandCategorical(arma::vec &weights) {
```

Comment:
Realistically you could use `DiscreteDistribution` here instead of adding something new.